### ISSUE TITLE
unresolved import for proc-macro generated identifier

### ISSUE THREAD

**Issue by ndmitchell** [rust-analyzer/rust-analyzer#6054]
In our project we have a bunch of types generated from proc-macros. Recently (probably since #6016) all such imports are flagged as unresolved import. That seems to be an expected consequence. Is there any way to disable such errors, as it makes it vastly harder to see real errors from Rust Analyzer.

**Comment by bjorn3** [comment]
`"rust-analyzer.diagnostics.disabled": ["unresolved-import"]` should work I think.

**Comment by bjorn3** [comment]
Also you can enable proc-macro support using

```jsonc
{
    "rust-analyzer.cargo.loadOutDirsFromCheck": true,
    "rust-analyzer.procMacro.enable": true,
}
```

**Comment by andrewwhitehead** [comment]
This just started showing up for me when importing types from the `typenum` crate. Strangely, if I change it to an unused import (`use typenum::U16` when only `U32` is referenced), that error seems to take priority over the unresolved type error.

**Comment by ecton** [comment]
I just started having this show up with `use serde_derive::{Serialize, Deserialize};` -- given Serde's popularity, I suspect many others will run into this one, and the main reason for my comment is to try to help make it findable.

@bjorn3's suggestion of enabling procMacro support worked for me to alleviate the issue.

**Comment by jonas-schievink** [comment]
> Is there any way to disable such errors, as it makes it vastly harder to see real errors from Rust Analyzer.

In the settings, add `unresolved-import` to the list of disabled diagnostics

**Comment by Veetaha** [comment]
I confirm that proc-macro-derives stopped working for me recently, not sure when exactly, but there should be some definite regression. All the necessary configurations are in place (`loadOutDirsFromCheck` and `procMacro.enable`):
![image](https://user-images.githubusercontent.com/36276403/94794467-914b7380-03e4-11eb-9fab-aa31c2af2f23.png)

**Comment by lnicola** [comment]
@Veetaha are you using nightly?

**Comment by Veetaha** [comment]
Yes, I always do

**Comment by lnicola** [comment]
https://github.com/rust-analyzer/rust-analyzer/pull/5651

Oh, it looks like I need to rebase it.

**Comment by spikespaz** [comment]
I'm getting this, not sure if it's related but when I search "rust-analyzer unresolved import on macro" this is where I'm sent.
```rust
#[cfg(test)]
mod tests {
    use test_case::test_case;
```
Not breaking anything, but a little annoying. Tips?

**Comment by BrandonDyer64** [comment]
Why was this issue closed? It's still very much a problem.

Also, setting `loadOutDirsFromCheck` and `procMacro.enable` doesn't help.

**Comment by lnicola** [comment]
@BrandonDyer64 because the root cause here (a change to the proc macro ABI) was fixed in https://github.com/rust-analyzer/rust-analyzer/pull/5651. It affected nightlies from the Rust 1.47. This was not the first proc macro interface change and it wasn't the last, but it makes no sense to track all of them in a single issue.

If macros are not working for you , please file a new one, with more information (compiler version, test case, whether no macros work at all, or some of them do etc.).

**Comment by BrandonDyer64** [comment]
@lnicola The trouble is though, that my issue would be literally this issue. It's exactly the same problem. I have an attribute proc macro that goes over a function and generates a struct. The resulting struct is not seen by rust-analyzer and is shown as an unresolved import.

**Comment by lnicola** [comment]
It can't be th same problem unless you're using a post-1.46 toolchain with a two year-old rust-analyzer (in which case you should upgrade).

So you should provide the info I mentioned before.

**Comment by BrandonDyer64** [comment]
Just realized I'm stuck on rustc 1.45, so never mind I guess
